php_mysqlnd_azure.c

```diff
@@ -34,30 +34,18 @@ static PHP_GINIT_FUNCTION(mysqlnd_azure)
 #if defined(COMPILE_DL_MYSQLND_AZURE) && defined(ZTS)
 	ZEND_TSRMLS_CACHE_UPDATE();
 #endif
-	mysqlnd_azure_globals->enabled = 0;
-	mysqlnd_azure_globals->redirectCache = NULL;
-}
-/* }}} */
-
-/* {{{ PHP_GSHUTDOWN_FUNCTION */
-static PHP_GSHUTDOWN_FUNCTION(mysqlnd_azure)
-{
-	if (mysqlnd_azure_globals->redirectCache) {
-		zend_hash_destroy(mysqlnd_azure_globals->redirectCache);
-		mnd_pefree(mysqlnd_azure_globals->redirectCache, 1);
-		mysqlnd_azure_globals->redirectCache = NULL;
-	}
+	mysqlnd_azure_globals->enableRedirect = 0;
 }
 /* }}} */
 
 /* {{{ PHP_INI */
 /*
 	It is handy to allow users to disable any mysqlnd plugin globally - not only for debugging :-)
-	Because we register our plugin in MINIT changes to mysqlnd_ed.enabled shall be bound to
+	Because we register our plugin in MINIT changes to myqlnd_azure.enableRedirect shall be bound to
 	INI_SYSTEM (and PHP restarts).
 */
 PHP_INI_BEGIN()
-STD_PHP_INI_ENTRY("mysqlnd_azure.enabled", "0", PHP_INI_ALL, OnUpdateBool, enabled, zend_mysqlnd_azure_globals, mysqlnd_azure_globals)
+STD_PHP_INI_ENTRY("mysqlnd_azure.enableRedirect", "0", PHP_INI_ALL, OnUpdateBool, enableRedirect, zend_mysqlnd_azure_globals, mysqlnd_azure_globals)
```

Comment:
Especially regarding the comment above, shouldn't this rather be `PHP_INI_SYSTEM` or `PHP_INI_PERDIR` instead of `PHP_INI_ALL`? See https://www.php.net/manual/en/configuration.changes.modes.php for the description of these modes.

Reply:
Hi Christoph,

Thanks a lot for your detailed check. The comment and code was inconsistent before. And after reconsideration, especially for that the option now cannot be compatiable with all use case, I think it may still leave experienced user an opportunity to manipulate it for their purpose. This will not cause harm, is this right?

Reply:
It seems that the cache is initialized lazily, so indeed switching the INI setting during request time should be fine.

Reply:
Thanks, I'll resolve the conversation then